apps/test/unit/templates/sectionProgress/progressTables/ProgressTableDetailCellTest.jsx

```diff
@@ -0,0 +1,143 @@
+    789: {pages: null, paired: false, result: 100, status: 'locked'}
+  },
+  stageExtrasEnabled: false
+};
+
+describe('ProgressTableDetailCell', () => {
+  beforeEach(() => {
+    sinon.stub(firehoseClient, 'putRecord');
+  });
+
+  afterEach(() => {
+    firehoseClient.putRecord.restore();
+  });
+
+  it('displays a progress table level bubble for each level and sublevel', () => {
+    const wrapper = shallow(<ProgressTableDetailCell {...DEFAULT_PROPS} />);
+    const levelBubble1 = wrapper.findWhere(node => node.key() === '123_1');
+    const levelBubble2 = wrapper.findWhere(node => node.key() === '456_2');
+    const levelBubble3 = wrapper.findWhere(node => node.key() === '999_3');
+    expect(levelBubble1.find(ProgressTableLevelBubble)).to.have.length(1);
+    expect(levelBubble2.find(ProgressTableLevelBubble)).to.have.length(2); // 1 for level, 1 for sublevel
+    expect(levelBubble3.find(ProgressTableLevelBubble)).to.have.length(1);
+  });
+
+  it('disables progress bubble if there is a bonus and stageExtrasEnabled is false', () => {
+    const wrapper = shallow(<ProgressTableDetailCell {...DEFAULT_PROPS} />);
+    const levelBubble3 = wrapper.findWhere(node => node.key() === '999_3');
+    const progressBubble = levelBubble3.find(ProgressTableLevelBubble);
+    expect(progressBubble.props().disabled).to.equal(true);
+  });
+
+  it('disable is false for progress bubble if there is a bonus and stageExtrasEnabled is true', () => {
+    const props = {...DEFAULT_PROPS, stageExtrasEnabled: true};
+    const wrapper = shallow(<ProgressTableDetailCell {...props} />);
+    const levelBubble3 = wrapper.findWhere(node => node.key() === '999_3');
+    const progressBubble = levelBubble3.find(ProgressTableLevelBubble);
+    expect(progressBubble.props().disabled).to.equal(false);
+  });
+
+  it('generates the right url for level bubble', () => {
+    const wrapper = shallow(<ProgressTableDetailCell {...DEFAULT_PROPS} />);
+    const levelBubble1 = wrapper.findWhere(node => node.key() === '123_1');
+    const url = levelBubble1.find(ProgressTableLevelBubble).props().url;
+    expect(url).to.equal('/level-1?section_id=123&user_id=1');
+  });
+
+  it('calls firehouse putRecord when clicking a level', () => {
+    const wrapper = shallow(<ProgressTableDetailCell {...DEFAULT_PROPS} />);
+    const levelBubble1 = wrapper
+      .findWhere(node => node.key() === '123_1')
+      .childAt(0);
+    levelBubble1.simulate('click');
+    expect(firehoseClient.putRecord).to.have.been.called;
+  });
+
+  it('calls firehouse putRecord when clicking a sublevel', () => {
+    const wrapper = shallow(<ProgressTableDetailCell {...DEFAULT_PROPS} />);
+    const sublevel = wrapper.findWhere(
+      node => node.key() === `${SUBLEVEL_1.id}`
+    );
+    sublevel.simulate('click');
+    expect(firehoseClient.putRecord).to.have.been.called;
+  });
+
+  it('widthForLevels returns the correct width when there are unplugged levels', () => {
```

Comment:
could you add a test to make sure the unplugged bubble properly resizes for a different locale? i only ask because the unplugged bubble used to have a fixed size, but i updated it to (hopefully) size perfectly to the localized text.